### PR TITLE
Use MKLocalSearch instead of geocoder to get better search results.

### LINKED ISSUE #6088: Improve location search by using local search
At the moment the search results for the location view are very poor and only geographically and not show stores and other kinds of results.

### PR DESCRIPTION
**Pull request by SergioEstevao**
Fixes #6088 

To test:
 - Create or Edit a post
 - Tap on Options
 - Tap on Set Location
 - Search for places around your current location
 - see if results are better than before
 - Check if selection of result works ok.

Needs review: @aerych

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Post/Geolocation/PostGeolocationViewController.m [hunk 1]
```diff
@@ -323,7 +326,7 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
             } else {
                 [self.geoView setCoordinate:coordinate];
             }
-            self.geoView.address = placemark.name;
+            self.geoView.address = [NSString stringWithFormat:@"%@, %@]",placemark.name, placemark.formattedAddress];
```
Comment (aerych) [inline WordPress/Classes/ViewRelated/Post/Geolocation/PostGeolocationViewController.m]: Super nitpicky, but maybe add a space before `placemark.name,`

**Comment by aerych** [comment]
Awesome change @SergioEstevao.  It really makes a huge difference in the search results I'm seeing when testing. 

I had one super nitpicky comment when looking at the code, but apart from that we probably need to tweak cell height in the search vc a bit as things are looking a bit crowded.  Here's an example from the simulator:

simulator screen shot oct 18 2016 3 18 30 pm

**Comment by SergioEstevao** [comment]
@aerych ready for another look! Thanks!

**Comment by aerych** [comment]
Looks good! :shipit:

**Pull request merged**